controller/node_controller.go

```diff
@@ -1954,3 +1859,117 @@ func isNodeOrDisksEvictionRequested(node *longhorn.Node) bool {
+				nc.eventRecorder, node, corev1.EventTypeNormal)
+		}
+	}()
+
+	nodeManagerFound := false
+	for _, pod := range managerPods {
+		if pod.Spec.NodeName == node.Name {
+			nodeManagerFound = true
+			podConditions := pod.Status.Conditions
+			for _, podCondition := range podConditions {
+				if podCondition.Type == corev1.PodReady {
+					if podCondition.Status != corev1.ConditionTrue || pod.Status.Phase != corev1.PodRunning {
+						nodeReady = false
+						node.Status.Conditions = types.SetConditionAndRecord(node.Status.Conditions,
+							longhorn.NodeConditionTypeReady, longhorn.ConditionStatusFalse,
+							string(longhorn.NodeConditionReasonManagerPodDown),
+							fmt.Sprintf("Node %v is down: the manager pod %v is not running", node.Name, pod.Name),
+							nc.eventRecorder, node, corev1.EventTypeWarning)
+					}
+					break
+				}
+			}
+			break
+		}
+	}
+	if !nodeManagerFound {
+		nodeReady = false
+		node.Status.Conditions = types.SetConditionAndRecord(node.Status.Conditions,
+			longhorn.NodeConditionTypeReady, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonManagerPodMissing),
+			fmt.Sprintf("manager pod missing: node %v has no manager pod running on it", node.Name),
+			nc.eventRecorder, node, corev1.EventTypeWarning)
+	}
+
+	kubeConditions := kubeNode.Status.Conditions
+	for _, con := range kubeConditions {
+		switch con.Type {
+		case corev1.NodeReady:
+			if con.Status != corev1.ConditionTrue {
+				if con.Status == corev1.ConditionFalse &&
+					time.Since(con.LastTransitionTime.Time) < ignoreKubeletNotReadyTime {
+					// When kubelet restarts, it briefly reports Ready == False. Responding too quickly can cause
+					// undesirable churn. See https://github.com/longhorn/longhorn/issues/7302 for an example.
+					nc.logger.Warnf("Ignoring %v == %v condition due to %v until %v", corev1.NodeReady, con.Status,
+						con.Reason, con.LastTransitionTime.Add(ignoreKubeletNotReadyTime))
+				} else {
+					nodeReady = false
+					node.Status.Conditions = types.SetConditionAndRecord(node.Status.Conditions,
+						longhorn.NodeConditionTypeReady, longhorn.ConditionStatusFalse,
+						string(longhorn.NodeConditionReasonKubernetesNodeNotReady), // This one.
+						fmt.Sprintf("Kubernetes node %v not ready: %v", node.Name, con.Reason),
+						nc.eventRecorder, node, corev1.EventTypeWarning)
+				}
+			}
+		case corev1.NodeDiskPressure,
+			corev1.NodePIDPressure,
+			corev1.NodeMemoryPressure,
+			corev1.NodeNetworkUnavailable:
+			if con.Status == corev1.ConditionTrue {
+				nodeReady = false
+				node.Status.Conditions = types.SetConditionAndRecord(node.Status.Conditions,
+					longhorn.NodeConditionTypeReady, longhorn.ConditionStatusFalse,
+					string(longhorn.NodeConditionReasonKubernetesNodePressure),
+					fmt.Sprintf("Kubernetes node %v has pressure: %v, %v", node.Name, con.Reason, con.Message),
+					nc.eventRecorder, node, corev1.EventTypeWarning)
+			}
+		}
+	}
```

Comment:
nit: abstract the codes

Reply:
Please check the latest commit to see if it meets your request.